tests/lib/totp.js

```diff
@@ -0,0 +1,143 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern!tdd',
+  'intern/chai!assert',
+  'tests/addons/environment',
+  'tests/addons/sinon',
+  'node_modules/otplib/otplib-browser'
+], function (tdd, assert, Environment, sinon, otplib) {
+
+  with (tdd) {
+    suite('totp', function () {
+      var account;
+      var accountHelper;
+      var respond;
+      var client;
+      var RequestMocks;
+      var env = new Environment();
+      var xhr;
+      var xhrOpen;
+      var xhrSend;
+
+      beforeEach(function () {
+        env = new Environment();
+        accountHelper = env.accountHelper;
+        respond = env.respond;
+        client = env.client;
+        RequestMocks = env.RequestMocks;
+
+        return accountHelper.newVerifiedAccount()
+          .then(function (newAccount) {
+            account = newAccount;
+            return respond(client.createTotpToken(account.signIn.sessionToken), RequestMocks.createTotpToken);
+          })
+          .then(function (res) {
+            assert.ok(res.qrCodeUrl, 'should return QR code data encoded url');
+            assert.ok(res.secret, 'should return secret that is encoded in url');
+
+            otplib.authenticator.options = {
+              secret: res.secret
+            };
```

Comment:
IIUC this `otplib.authenticator` is mutable state shared between tests; could we use ` new otplib.authenticator.Authenticator()` to ensure that the tests don't accidentally share state?